tests/e2e/v2store_deprecation_test.go

```diff
@@ -250,7 +251,23 @@ func assertSnapshotsMatch(t testing.TB, firstDataDir, secondDataDir string, patc
 		if err != nil {
 			t.Fatal(err)
 		}
-		assert.Equal(t, openSnap(patch(firstSnapshot.Data)), openSnap(patch(secondSnapshot.Data)))
+		assertMembershipEqual(t, openSnap(patch(firstSnapshot.Data)), openSnap(patch(secondSnapshot.Data)))
```

Comment:
Why change this? Doesn't seem related changing the logic, looks like additional debug info? Could you split unrelated changes to PR?

Reply:
> The events on v2store and indices on v2store wont match between the two snapshots anymore

Not sure I understand. It's not clear from code changes. Please separate a PR that updates the test. PR should be organized in a way to clearly show what is changing, if we need to change validation, let's do that separately from business logic.

Reply:
Adding more clarification on the test change -
Original test was checking that the v2 snapshot content created by older and newer version matched wrt indices on the v2store and events on the v2 store. With the new logic, the indices and events on v2store will not match. As a result we saw a test failure. The test is being updated to verify membership match instead. Please let me know if you have more questions. I would like to confirm that we are on same page wrt this change in the test.

Created separate PR for test changes as per review feedback above (https://github.com/etcd-io/etcd/pull/16441) . The test change will need to  be merged before this PR can pass. Converting this one to draft and removing the test change. The test will start failing until the test PR merges.

Reply:
Do not understand why we have to create a separate PR to change the validation. We only care about the membership data in the snapshot, it's reasonable to only compare the membership data.

Anyway, approved the other PR https://github.com/etcd-io/etcd/pull/16441.

Reply:
Thanks both. Separate CR was used for test code as per review feedback.